MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -1296,32 +1310,36 @@ private static Video[] LoadThemeVideos(IEnumerable<FileSystemMetadata> fileSyste
 
         protected virtual BaseItem[] LoadExtras(List<FileSystemMetadata> fileSystemChildren, IDirectoryService directoryService)
         {
-            var files = fileSystemChildren.Where(i => i.IsDirectory)
-                .SelectMany(i => FileSystem.GetFiles(i.FullName));
+            var extras = new List<Video>();
 
-            return LibraryManager.ResolvePaths(files, directoryService, null, new LibraryOptions())
-                .OfType<Video>()
-                .Select(item =>
-                {
-                    // Try to retrieve it from the db. If we don't find it, use the resolved version
-                    var dbItem = LibraryManager.GetItemById(item.Id) as Video;
+            foreach (var extraFolderName in AllExtrasTypesFolderNames)
+            {
+                var files = fileSystemChildren.Where(i => i.IsDirectory)
```

Comment:
This is executed for each folder, maybe move it out the loop?
Or use `IEnumerable.Contains` instead of a `foreach` loop and `string.Equals`